wolfSSL-Porting/src/section02.md

```diff
@@ -127,11 +127,225 @@ If wolfSSL needs to be used in a multithreaded environment, the wolfSSL mutex la
+```
+
+Where `myHardwareRNG` is a function with the signature:
+```c
+int myHardwareRNG(unsigned char* output, unsigned int sz);
+```
+
+This function should fill the `output` buffer with `sz` bytes of random data from your hardware RNG and return 0 on success.
+
+### Custom Seed Generation Methods
+
+wolfSSL provides several methods to customize how the DRBG obtains its seed:
+
+#### 1. CUSTOM_RAND_GENERATE_SEED
+
+Define a custom function that generates seed data. This is the most direct way to provide entropy.
+
+```c
+#define CUSTOM_RAND_GENERATE_SEED myGenerateSeed
+
+int myGenerateSeed(unsigned char* output, unsigned int sz)
+{
+    // Fill output buffer with sz bytes of entropy from your source
+    // Return 0 on success, non-zero on error
+}
+```
+
+#### 2. CUSTOM_RAND_GENERATE
+
+Define a function that returns random values one at a time. wolfSSL will call this function repeatedly to fill the seed buffer.
+
+```c
+#define CUSTOM_RAND_GENERATE myRandFunc
+#define CUSTOM_RAND_TYPE unsigned int
+
+unsigned int myRandFunc(void)
+{
+    // Return a random value from your entropy source
+}
+```
+
+The `CUSTOM_RAND_TYPE` should match the return type of your function (e.g., `unsigned int`, `unsigned long`, etc.).
+
+Example from `./wolfcrypt/src/random.c` (lines 2707-2736):
+```c
+int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
+{
+    word32 i = 0;
+    while (i < sz) {
+        if ((i + sizeof(CUSTOM_RAND_TYPE)) > sz ||
+            ((wc_ptr_t)&output[i] % sizeof(CUSTOM_RAND_TYPE)) != 0) {
+            output[i++] = (byte)CUSTOM_RAND_GENERATE();
+        }
+        else {
+            *((CUSTOM_RAND_TYPE*)&output[i]) = CUSTOM_RAND_GENERATE();
+            i += sizeof(CUSTOM_RAND_TYPE);
+        }
+    }
+    return 0;
+}
+```
+
+#### 3. wc_SetSeed_Cb (Runtime Callback)
+
+Set a seed generation callback at runtime using `wc_SetSeed_Cb()`. This requires defining `WC_RNG_SEED_CB` at build time.
+
+Build configuration:
+```c
+#define WC_RNG_SEED_CB
+```
+
+Runtime usage:
+```c
+#include <wolfssl/wolfcrypt/random.h>
+
+int myCustomSeedFunc(OS_Seed* os, byte* seed, word32 sz)
+{
+    // Generate sz bytes of entropy into seed buffer
+    // Return 0 on success
+}
+
+// Set the callback before initializing RNG
+wc_SetSeed_Cb(myCustomSeedFunc);
+
+// Now initialize and use RNG as normal
+WC_RNG rng;
+wc_InitRng(&rng);
+```
+
+The callback function is defined in `./wolfcrypt/src/random.c` (lines 319-323).
+
+#### 4. Crypto Callbacks (WC_ALGO_TYPE_SEED)
+
+Use wolfSSL's crypto callback mechanism to provide seed generation through a hardware security module or custom crypto device. This requires `WOLF_CRYPTO_CB` to be defined.
+
+Build configuration:
+```c
+#define WOLF_CRYPTO_CB
+```
+
+Implementation example (from `wolfssl-examples/tls/cryptocb-common.c`):
+```c
+int myCryptoCb(int devIdArg, wc_CryptoInfo* info, void* ctx)
+{
+    if (info->algo_type == WC_ALGO_TYPE_SEED) {
+        // Generate random seed data
```

Comment:
Should we use `/* */` style comments in example code (here and other places in this PR) to match our coding style?